### PR TITLE
Added GSQL lexer

### PR DESCRIPTION
**Pull request by DanBarkus**
Added lexer for GSQL language and handling .gsql files

### CODE REVIEW

File: pygments/lexers/gsql.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            include('whitespace'),
+            include('barewords'),
+            include('operators'),
+        ],
+        'comment': [
+            (r'.*\#.*\n', Comment.Single),
+            (r'.*\/\*\s*.*\s*\*\/', Comment.Multiline),
+        ],
+        'keywords': [
+            (r'(ACCUM|AND|ANY|API|AS|ASC|AVG|BAG|BATCH|BETWEEN|BOOL|BOTH|'
```
Comment (Anteru) [inline pygments/lexers/gsql.py:34]: Please use the `words` function for long lists of keywords which can optimize the matching, see for instance: https://github.com/pygments/pygments/blob/master/pygments/lexers/python.py#L202

File: pygments/lexers/gsql.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import re
+
+from pygments.lexer import RegexLexer, include, bygroups, using, this
+from pygments.token import Keyword, Punctuation, Comment, Operator, Name,\
+    String, Number, Whitespace
+
+
+__all__ = ["GSQLLexer"]
+
+class GSQLLexer(RegexLexer):
```
Comment (Anteru) [inline pygments/lexers/gsql.py:10]: Please add a comment describing this lexer -- ideally with a link to the language definition. This comment should end with `.. versionadded:: 2.10`

**Review by Anteru — CHANGES_REQUESTED** [review]

**Comment by DanBarkus** [comment]
All requested changes have been made.

File: pygments/lexers/gsql.py (hunk 2 above)
Comment (zrougamed) [reply pygments/lexers/gsql.py:10]: @DanBarkus thanks for the great work :smile: 
@Anteru I can't wait to see this PR Merged :smile:

**Comment by HerkTG** [comment]
@DanBarkus  thanks for completing this!!
@Anteru It would be great if you could review. We are looking to use this for an O'Reilly Book. Please review as soon as possible. Thanks!

File: pygments/lexers/gsql.py (hunk 2 above)
Comment (GenericP3rson) [reply pygments/lexers/gsql.py:10]: @DanBarkus Very cool! Can't wait to use it!
@Anteru Looking forward to the merge!

File: pygments/lexers/gsql.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            (words(('accum', 'having', 'limit', 'order', 'postAccum', 'sample', 'where')), Name.Builtin)
+        ],
+        'accums': [
+            (words(('andaccum', 'arrayaccum', 'avgaccum', 'bagaccum', 'bitwiseandaccum', 
+             'bitwiseoraccum', 'groupbyaccum', 'heapaccum', 'listaccum', 'MapAccum', 
+             'maxaccum', 'minaccum', 'oraccum', 'setaccum', 'sumaccum')), Name.Builtin),
+        ],
+        # 'name': [
+        #     (r'(\@\@w+)\b', Name),
+        # ],
```
Comment (Anteru) [inline pygments/lexers/gsql.py:63]: Can you please remove this?

**Comment by Anteru** [comment]
Merged, thanks! I'd appreciate if you could do a follow up commit to trim the example file a bit and reply to the question about `name` being commented out.

**Comment by Anteru** [comment]
I also had to regenerate the golden test files, could you please check that `Name.Builtin` is acceptable for things like `FOR` which were `Keyword` previously?

**Comment by Anteru** [comment]
@DanBarkus This is also failing the docs build as it's missing the top-level comment which includes the copyright annotation. I'm reverting the PR for now until this is addressed. Please reopen once this is resolved.

**Comment by Anteru** [comment]
Reverted via 0a93d6c8ebe20d0e90883f210cd2db09f990ce0d and 066bd7e5f4bf864a3deac5994fa37f79801b2d23

**Comment by DanBarkus** [comment]
I've added the missing copyright annotation and removed the unnecessary commented out section of code.
I reviewed uses of the words function in other modules (sql, python) and usage patterns line up with those languages.
Will submit new pull request with updated code.

**Pull request merged**